src/core/Pills/BasicPills/BasicPills.tsx

```diff
@@ -0,0 +1,18 @@
+import React, { useState } from 'react';
+import { useStyles } from './style';
+import { Button } from '@material-ui/core';
+
+type selectType = true | false;
+const BasicPills: React.FC = ({ children }) => {
+  const classes = useStyles();
+  const [selected, setSelected] = useState<selectType>(false);
+  return (
+    <Button
+      onClick={() => setSelected(!selected)}
+      className={`${classes.root} ${selected ? classes.selected : ''}`}
+    >
+      {children}
+    </Button>
```

Comment:
I think the pills shouldn't have an onClick toggler. It has two states, `default` - when not selected and `active` - when selected

When the pill is normal `root` class will have base styling
When the pills are selected `selected` property of the button would be overridden. (Just like hover)

Basically, pills would never be clicked/clickable, they can either take a prop as `selected` or not, or you could follow an overridden approach based on a div. Wherever the pills are attached (a card, template, etc) if that particular div/box is selected the pills should be active (not by directly clicking on the pill).